### PR TITLE
Openapi v2 swagger generation annotations (#2)

### PR DESCRIPTION
**Pull request by ab666**
- Add ability to exclude http routes from swagger
- Swagger operation options added
- Added swagger root info

### CODE REVIEW

File: swagger/src/main/proto/openapi_v2.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+extend google.protobuf.ServiceOptions {
+  Swagger root = 62000;
+}
+
+extend google.protobuf.MethodOptions {
+  Operation operation = 72000;
+  bool exclude = 72001; // indicates that the method should be excluded from swagger generation.
+}
+
+
```
Comment (akalini) [inline swagger/src/main/proto/openapi_v2.proto]: nit: empty line :)

File: swagger/src/main/proto/openapi_v2.proto [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object
+message Swagger {
+  Info info = 1;
+  string host = 2;
+  string basePath = 3;
+}
+
+
```
Comment (akalini) [inline swagger/src/main/proto/openapi_v2.proto]: nit: empty line.

File: swagger/src/test/java/grpcbridge/swagger/PetstoreOpenapiV2Test.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                .addService(new PetStoreServiceImplBase(){}.bindService())
+                .build();
+    }
+
+    @Test
+    public void generateManifest_petstore_v2() {
+        OpenapiV2.Swagger swaggerRoot = descriptor.getServices().stream()
+                .findAny()
+                .orElseThrow(() -> new RuntimeException("No service defied"))
+                .getOptions().getExtension(OpenapiV2.root);
```
Comment (akalini) [inline swagger/src/test/java/grpcbridge/swagger/PetstoreOpenapiV2Test.java]: nit: .getExension on the next line.

File: swagger/src/test/java/grpcbridge/swagger/PetstoreOpenapiV2Test.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        String manifest = bridge.generateManifest(BridgeSwaggerManifestGenerator
+                .newBuilder()
+                .setSwaggerRoot(swaggerRoot)
+                .build()
+        );
```
Comment (akalini) [inline swagger/src/test/java/grpcbridge/swagger/PetstoreOpenapiV2Test.java]: What's this, Kotlin? :) Join with prev line please :)

File: swagger/src/main/java/grpcbridge/swagger/BridgeSwaggerManifestGenerator.java [hunk 5]
```diff
@@ -34,17 +37,39 @@ public String generate(List<Route> routes) {
...
                 .map(route -> route.descriptor)
                 .filter(descriptor -> descriptor.getOptions().hasExtension(AnnotationsProto.http))
+                .filter(descriptor -> !descriptor.getOptions().hasExtension(OpenapiV2.exclude)
+                        || !descriptor.getOptions().getExtension(OpenapiV2.exclude))
                 .forEach(descriptor -> applyMethodDescriptor(schema, descriptor));
 
         return schema.serialize();
     }
 
+    private <T> T getSwaggerFiled(Function<OpenapiV2.Swagger,T> extractor) {
```
Comment (sam-step) [inline swagger/src/main/java/grpcbridge/swagger/BridgeSwaggerManifestGenerator.java]: `getSwaggerField`?

**Comment by ab666** [comment]
All corrected, thanks!

**Pull request merged**